src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Device, Inc. All rights reserved.
+
+#include <string>
+#include "core/common/error.h"
+#include "hip/config.h"
+#include "hip/core/device.h"
+#include "hip/core/memory.h"
+#include "hip/hip_runtime_api.h"
+
+namespace xrt::core::hip
+{
+
+  // Allocate memory on the device.
+  static void
+  hip_malloc(void **ptr, size_t size)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Allocates device accessible host memory.
+  static void
+  hip_host_malloc(void **ptr, size_t size, unsigned int flags)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Free memory allocated by the hipHostMalloc().
+  static void
+  hip_host_free(void *ptr)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Free memory allocated by the hipMalloc().
+  static void
+  hip_free(void *ptr)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Register host memory so it can be accessed from the current device.
+  static void
+  hip_host_register(void *hostPtr,
+                  size_t sizeBytes,
+                  unsigned int flags)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Un-register host pointer.
+  static void
+  hip_host_unregister(void *hostPtr)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Get Device pointer from Host Pointer allocated through hipHostMalloc().
+  static void
+  hip_host_get_device_pointer(void **devPtr,
+                          void *hstPtr,
+                          unsigned int flags)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+  // Copy data from src to dst.
+  static void
+  hip_memcpy(void *dst,
+            const void *src,
+            size_t sizeBytes,
+            hipMemcpyKind kind)
+  {
+    throw std::runtime_error("Not implemented");
+  }
+
+} // xrt::core::hip
+
+// Allocate memory on the device.
+hipError_t
+hipMalloc(void **ptr, size_t size)
+{
+  if (size == 0)
+  {
+    *ptr = nullptr;
+    return hipSuccess;
+  }
+  try  {
```

Comment:
Small nit, single space after "try". Please make changes in other places too.

Reply:
Done.